### PR TITLE
Refactor out membership engagement banner template into it's own file, use getSync for geolocaiton

### PR DESCRIPTION
**Pull request by jranks123**
## What does this change?
In order to create tests with new banner templates more easily, this change makes it simple for a test to provide an alternative engagement banner template.

Also, I am separating the message main text from the cta in the EngagementBannerParams; this is because in an upcoming test, one variant makes the main paragraph big and keeps the cta small, so in order to generalise across the tempaltes they need splitting out

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/templates/acquisitions-banner-control.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    buttonCaption = '',
+    buttonSvg = '',
+}: {
+    messageText: string,
+    paypalAndCreditCardImage: string,
+    colourClass: string,
+    linkUrl: string,
+    buttonCaption: string,
+    buttonSvg: string,
+}) =>
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/commercial/templates/acquisitions-banner-control.js]: Could this be of type EngagementBannerTemplateParams?

File: .gitignore [hunk 2]
```diff
@@ -72,3 +72,4 @@ static/src/stylesheets/pasteup/.npmrc
 *.crt
 *.key
 .vscode/settings.json
+.yarnrc
```
Comment (SiAdcock) [inline .gitignore]: Was this automatically added by an IDE?

Comment (SiAdcock) [reply .gitignore]: Sorry, just read your parenthetical! You could add this to your global gitignore if it is specific for your machine

**Review by gu-stav — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner-parameters.js [hunk 3]
```diff
@@ -61,20 +61,19 @@ const monthlySupporterCost = (location: string): string => {
...
 
 const supporterParams = (location: string): EngagementBannerParams =>
     Object.assign({}, baseParams, {
         buttonCaption: 'Become a Supporter',
         linkUrl: 'https://membership.theguardian.com/supporter',
         products: ['MEMBERSHIP_SUPPORTER'],
-        messageText: supporterEngagementBannerCopy(location),
+        messageText: engagementBannerCopy(),
+        ctaText: supporterEngagementCtaCopy(location),
         pageviewId: (config.ophan && config.ophan.pageViewId) || 'not_found',
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner-parameters.js]: Optional: this assignment can be replaced with:
```js
pageviewId: config.get('ophan.pageViewId', 'not_found'),
```
Same on line 87

**Review by SiAdcock — APPROVED** [review]
I've added an optional suggestion to improve existing code, but your changes are good so feel free to merge regardless 😄

File: static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner-parameters.js (hunk 3 above)
Comment (jranks123) [reply static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner-parameters.js]: nice!

**Pull request merged**